### PR TITLE
Adding Reqmodule

### PR DESCRIPTION
**Pull request by melyxlin**
This PR adds the UI of the Request Module for the web app.
Includes:
- a place for users to add the headers, request body, HTTP method, and token used to get the request

### CODE REVIEW

File: web/client/src/components/request/ReqModule.jsx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    }
+  };
+
+  const handleClose = (num) => {
+    if (num === 1) {
+      setOpen1(false);
+    } else if (num === 2) {
+      setOpen2(false);
+    }
+  };
```
Comment (LeoRuiz01) [inline web/client/src/components/request/ReqModule.jsx:47]: I don't think you need to have a handleClickOpen or a handleClose function. You can just call onClick = {() => setOpen1(true)}. A similar thing could be done for the other button by calling setOpen2.

File: web/client/src/components/request/ReqModule.jsx [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+import DeleteIcon from "@material-ui/icons/Delete";
+import "../../styles/request.css";
+import { object, string } from "yup";
+
+/**
+ * @return {Formik} component using Formik for creating send Request
+ */
+export default function ReqModule() {
+  const [open1, setOpen1] = useState(false);
+  const [open2, setOpen2] = useState(false);
```
Comment (LeoRuiz01) [inline web/client/src/components/request/ReqModule.jsx:31]: You could change the names of these variables so they represent the dialog section they are opening just to make it more clear what it is being used for.

Comment (melyxlin) [reply web/client/src/components/request/ReqModule.jsx:31]: Fixed, but idk why you can't see the commit on the PR

File: web/client/src/components/request/ReqModule.jsx (hunk 1 above)
Comment (melyxlin) [reply web/client/src/components/request/ReqModule.jsx:47]: Fixed, but idk why you can't see the commit on the PR

File: web/client/src/components/request/ReqModule.jsx (hunk 2 above)
Comment (LeoRuiz01) [reply web/client/src/components/request/ReqModule.jsx:31]: Apparently github is having issues, https://www.githubstatus.com/

Comment (LeoRuiz01) [reply web/client/src/components/request/ReqModule.jsx:31]: Github is having some issues,https://www.githubstatus.com/

File: web/client/src/App.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ import {
   TokenForm,
   ValidateToken,
   TokenDisplay,
+  ReqModule,
```
Comment (yousefa00) [inline web/client/src/App.js:7]: Use full name for components: `RequestModule`

File: web/client/src/App.js [hunk 4]
```diff
@@ -75,9 +76,18 @@ function App() {
           </Grid>
           <Grid item xs>
             <MaterialUI paperClass="paper-bottom">
-              <ModuleInfo title="" content={[]} hasNote={false} note="" />
+              <ModuleInfo
+                title="Using a Token in a HTTP Request"
```
Comment (yousefa00) [inline web/client/src/App.js:80]: Change capitalization to match label format. I also suggest: Use your token for an HTTP request

File: web/client/src/components/request/ReqModule.jsx (hunk 2 above)
Comment (yousefa00) [reply web/client/src/components/request/ReqModule.jsx:31]: Seconded, these variable names should be changed

File: web/client/src/components/request/ReqModule.jsx [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+  const handleClickOpen = (num) => {
+    if (num === 1) {
```
Comment (yousefa00) [inline web/client/src/components/request/ReqModule.jsx:34]: Use a more descriptive name for num

File: web/client/src/styles/request.css [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.request-content {
+  margin-top: 2rem !important;
```
Comment (yousefa00) [inline web/client/src/styles/request.css:2]: Remove `!important`

File: web/client/src/App.js (hunk 3 above)
Comment (melyxlin) [reply web/client/src/App.js:7]: Yep changed, but GitHub is down so cannot see it

File: web/client/src/components/request/ReqModule.jsx (hunk 5 above)
Comment (melyxlin) [reply web/client/src/components/request/ReqModule.jsx]: Changed

File: web/client/src/styles/request.css (hunk 6 above)
Comment (melyxlin) [reply web/client/src/styles/request.css:2]: important needs to be here or there will be no spacing between the parts

**Comment by melyxlin** [comment]
I am going to make a new PR with the new changes and just merge it from there

**Pull request closed without merging**